src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -291,6 +287,24 @@ public virtual Task DispatchEventAsync(ulong eventHandlerId, EventFieldInfo? fie
             return result;
         }
 
+        /// <summary>
+        /// Gets the event arguments type for the specified event handler.
+        /// </summary>
+        /// <param name="eventHandlerId">The <see cref="RenderTreeFrame.AttributeEventHandlerId"/> value from the original event attribute.</param>
+        /// <returns>The parameter type expected by the event handler. Normally this is a subclass of <see cref="EventArgs"/>.</returns>
+        public Type GetEventArgsType(ulong eventHandlerId)
```

Comment:
Is the reason for this being public that we need to call it from other assemblies?

Reply:
Yes, and because it's really becoming a part of the contract a renderer makes with its host. The web host has reasons for wanting to know the .NET type to supply for an event argument, so other hosts may want to do that too.

Reply:
I'm unclear why we need to do this based on the event handler id vs based on the event name since there can't be two registrations for the same name.

What is preventing us from building a map at host initialization time of (eventName -> type) and using that for the deserialization?

Reply:
> What is preventing us from building a map at host initialization time of (eventName -> type) and using that for the deserialization?

That information doesn't exist in any formal way. The `[EventHandler]` registrations are a convention for the Razor compiler, but developers aren't compelled to use it.

The information that does exist for sure is the mapping from `eventHandlerId`->`Delegate`, which the renderer already stores.

Reply:
> That information doesn't exist in any formal way. The `[EventHandler]` registrations are a convention for the Razor compiler, but developers aren't compelled to use it.

Then the comment [here](https://github.com/dotnet/aspnetcore/pull/29993#discussion_r576939857) is not completely correct, nothing prevents me from using a random type as an event handler provided that it inherits from EventArgs.